src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -0,0 +1,704 @@
+/* eslint-disable no-throw-literal */
+/* eslint-disable no-console */
+/* eslint-disable no-continue */
+import { generateUUID } from 'react-native-database';
+import { NativeModules } from 'react-native';
+
+const SENSOR_LOG_PRE_AGGREGATE_TYPE = 'preAggregate';
+const SENSOR_LOG_FULL_AGGREGATE_TYPE = 'aggregate';
+const SENSOR_LOG_BREACH_AGGREGATE_TYPE = 'breachAggregate';
+
+const SENSOR_SYNC_COMMAND_GET_LOGS = '*logall';
+const SENOSOR_SYNC_COMMAND_RESET_INTERVAL = '*lint240'; // 4 minutes
+const SENSOR_SYNC_COMMAND_RESET_ADVERTISEMENT_INTERVAL = '*sadv1000';
+const SENSOR_SYNC_CONNECTION_DELAY = 450;
+const SENSOR_SYNC_NUMBER_OF_RECONNECTS = 11;
+
+const ONE_MINUTE_MILLISECONDS = 1000 * 60;
+const ONE_HOUR_MILLISECONDS = ONE_MINUTE_MILLISECONDS * 60;
+const EIGHT_HOURS_MILLISECONDS = ONE_HOUR_MILLISECONDS * 8;
+const FIVE_MINUTES_MILLISECONDS = ONE_MINUTE_MILLISECONDS * 5;
+const NO_FULL_AGGREGATE_PERIOD = EIGHT_HOURS_MILLISECONDS * 4 + FIVE_MINUTES_MILLISECONDS;
+// Include a small offset on the interval to account for each timestamp being
+// small amounts of time off.
+const FULL_AGGREGATION_INTERVAL = EIGHT_HOURS_MILLISECONDS + FIVE_MINUTES_MILLISECONDS;
+const fullInt = 256 * 256;
+const halfInt = fullInt / 2;
+const millisecondInMinute = 60 * 1000;
+const preAggregateInterval = 20 * millisecondInMinute;
+const manufacturerID = 307;
+const sensorScanTimeout = 10000;
+
+function toUInt(byteArray, startPosition) {
+  // TO DO negative temperatures (this is unsigned int conversion)
+  return byteArray[startPosition] * 256 + byteArray[startPosition + 1];
+}
+
+function toInt(byteArray, startPosition) {
+  const uINT = toUInt(byteArray, startPosition);
+  if (uINT > halfInt) {
+    return (fullInt - uINT) * -1;
+  }
+  return uINT;
+}
```

Comment:
This doesn't always give a correct result. 

Ideally all bit twiddling should be done in Java and a fully parsed object is returned.

Otherwise if we have to do this is JS, a more standard way of flipping the bits and adding one would be preferred to avoid any mistakes. (JS Number type I believe is 32 bits, so a bit mask for the first 16 bits would need to be used)

Reply:
not doing